js/flightlog_fielddefs.js

```diff
@@ -251,9 +255,10 @@ var
     ]),
 
     FILTER_TYPE = makeReadOnly([
-            "PT1",
-            "BIQUAD",
-            "FIR",
```

Comment:
we are removing backward compatiblity for it?

Reply:
FIR was never implemented in firmware, and I'm only adding PT2 and PT3 to the enum... 
I don't know if this is OK or not, at this point, and am awaiting further advice.

Reply:
It does not depend if it was implemented or not... It depends on the value the firmware wrote to the log, as int, this enum must be consistent

Reply:
What I mean is that we have had PT1 and Biquad as the only options before.  So 0 was always PT1 and 1 always Biquad.   The only change now is to add PT2 and PT3 on the end as options 2 and 3.  These were not used previously, as far as I am aware.

Reply:
We can decide if version 3.3.3 needs to be supported or not. It seems it existed, and SLEW too:

https://github.com/betaflight/betaflight/blob/3565182d311e5151526f8bb6c2d8f3c9192cd735/src/main/common/filter.h#L65-L70

Reply:
My personal feeling is that we should not worry about 3.3.3, since it was:
- 'unusual' in its filter options, they were 'experimental', I think;
- FIR was rarely, if ever, used, in practice;
- it was a long time ago, and
- 3.5.7 is the '3' version that people fly nowadays, it has PT1 and biquad options only.
Since I don't know how to code enum version control, I would be grateful if I could leave the PR as is.
If you know a good way to provide version control over enums, and if you had the time to implement a PR that would provide this, I'm sure there will be occasions in future where we may find that useful, so I leave that for you to decide.

Reply:
While I agree with the statement that we should not worry about 3.3.3 when it comes to displaying everything correctly, I think there are some things to be considered:
- I think while people are not necessarily flying 3.3.3 any more, they still might have old logs from 3.3.3 around, and might want to open them with blackbox explorer.
- in this case displaying wrong data without at least a warning is misleading
What are thoughts around either adding a warning when opening old logs, telling the user that some data may be incorrect?

Reply:
Good point.  Maybe if a version of the log viewer doesn't fully logs before some firmware version, then we can show a message recommending to use an earlier version of the log viewer.  The message could be 'smart' and say which fields may not be displayed correctly, if we know.
Currently there is a lot of maintenance work required in the log viewer.